### PR TITLE
Test against Rails main, fix `activerecord-trilogy-adapter v3.1+`

### PR DESCRIPTION
**Pull request by HeyNonster**
- Adds a workflow to start testing this gem with Rails main on a schedule. I haven't added any notifications for that yet, just checking that it works.
- Removes the lockfiles and the upper limit on ActiveRecord.
- Starts testing Rails 6.1 with Trilogy now that the adapter supports it.
- Fixes using `activerecord-trilogy-adapter v3.1+`.
- Stops testing with Ruby 3.0.

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# .git-blame-ignore-revs
+# Fix StandardRB Style/StringLiterals offenses
+bb2db8fa836254c7b3f2ba0eb4313a9a5ba17bcd
```
Comment (bquorning) [inline .git-blame-ignore-revs:3]: Feel free to add eb96d02c2bce7d7cd4a8a35e6ce206d83bbcd2c1 also.

File: lib/active_record_host_pool.rb [hunk 2]
```diff
@@ -15,7 +15,11 @@ class << self
   ActiveRecordHostPool.loaded_db_adapter = :mysql2
 elsif Gem.loaded_specs.include?("trilogy")
   require "trilogy"
-  require "activerecord-trilogy-adapter" if ActiveRecord.version < Gem::Version.new("7.1")
+  if ActiveRecord.version < Gem::Version.new("7.1")
+    require "activerecord-trilogy-adapter"
+    require "trilogy_adapter/version"
+    raise "ActiveRecordHostPool is only compatible with activerecord-trilogy-adapter v3.1+" if Gem::Version.new("3.1") > TrilogyAdapter::VERSION
```
Comment (bquorning) [inline lib/active_record_host_pool.rb:21]:
```suggestion
    raise "ActiveRecordHostPool is only compatible with activerecord-trilogy-adapter v3.1+" if TrilogyAdapter::VERSION < Gem::Version.new("3.1")
```

File: active_record_host_pool.gemspec [hunk 3]
```diff
@@ -17,5 +17,5 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = ">= 3.0.0"
```
Comment (bquorning) [inline active_record_host_pool.gemspec:18]:
```suggestion
  s.required_ruby_version = ">= 3.1.0"
```

File: lib/active_record_host_pool.rb (hunk 2 above)
Comment (HeyNonster) [reply lib/active_record_host_pool.rb:21]: Standardrb made me switch this.

File: active_record_host_pool.gemspec (hunk 3 above)
Comment (HeyNonster) [reply active_record_host_pool.gemspec:18]: I wasn't sure if we wanted to drop the gem from bundling with Ruby 3.0 altogether or not but happy to require Ruby 3.1+ 🤷

File: lib/active_record_host_pool.rb (hunk 2 above)
Comment (bquorning) [reply lib/active_record_host_pool.rb:21]: What, standardrb says line 18 and 21 are both fine, even if they check in opposite order? It’s the YodaCondition cop, right?

Comment (HeyNonster) [reply lib/active_record_host_pool.rb:21]: It is:

```
  lib/active_record_host_pool.rb:21:96: Style/YodaCondition: Reverse the order of the operands `TrilogyAdapter::VERSION < Gem::Version.new("3.1")`.
```

File: active_record_host_pool.gemspec (hunk 3 above)
Comment (razumau) [reply active_record_host_pool.gemspec:18]: I would prefer to drop it completely, rather than allow an untested version.

**Review by razumau — APPROVED** [review]

**Pull request merged**